### PR TITLE
Worldgen Mappings

### PR DESCRIPTION
**Pull request by jaskarth**
This PR improves FoliagePlacer mappings, SurfaceBuilder mappings, and a lot of general method names and parameter names. It also adds some javadoc to FoliagePlacers, as they are rather complicated to understand.

### CODE REVIEW

File: mappings/net/minecraft/util/collection/WeightedList.mapping [hunk 1]
```diff
@@ -1,21 +1,48 @@
...
+		ARG 0 object
 	METHOD method_23333 shuffle (Ljava/util/Random;)Lnet/minecraft/class_4131;
 		ARG 1 random
+	METHOD method_23334 (Ljava/util/Random;Lnet/minecraft/class_4131$class_4132;)V
+		ARG 1 entry
 	METHOD method_23337 pickRandom (Ljava/util/Random;)Ljava/lang/Object;
 		ARG 1 random
+	METHOD method_28337 (Lnet/minecraft/class_4131;)Ljava/util/List;
+		ARG 0 list
+	METHOD method_28338 weightedListOf (Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
```
Comment (YanisBft) [inline mappings/net/minecraft/util/collection/WeightedList.mapping]:
```suggestion
	METHOD method_28338 createCodec (Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/util/collection/WeightedList.mapping [hunk 2]
```diff
@@ -1,21 +1,48 @@
...
 		FIELD field_18401 weight I
 		FIELD field_18402 shuffledOrder D
+		METHOD <init> (Ljava/lang/Object;I)V
+			ARG 1 item
+			ARG 2 weight
 		METHOD method_19033 getShuffledOrder ()D
 		METHOD method_19034 setShuffledOrder (F)V
 			ARG 1 random
 		METHOD method_19035 getElement ()Ljava/lang/Object;
+		METHOD method_28341 entryCodecOf (Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
```
Comment (YanisBft) [inline mappings/net/minecraft/util/collection/WeightedList.mapping]:
```suggestion
		METHOD method_28341 createCodec (Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/util/collection/WeightedList.mapping [hunk 3]
```diff
@@ -1,21 +1,48 @@
...
+				ARG 1 ops
+				ARG 2 object
+			METHOD encode (Ljava/lang/Object;Lcom/mojang/serialization/DynamicOps;Ljava/lang/Object;)Lcom/mojang/serialization/DataResult;
+				ARG 1 entry
+				ARG 2 ops
+				ARG 3 object
+			METHOD method_28344 (Lcom/mojang/serialization/Dynamic;Ljava/lang/Object;)Lnet/minecraft/class_4131$class_4132;
+				ARG 1 object
+			METHOD method_28345 (Lcom/mojang/serialization/DynamicOps;Lnet/minecraft/class_4131$class_4132;)Lcom/mojang/datafixers/util/Pair;
+				ARG 1 entry
```
Comment (YanisBft) [inline mappings/net/minecraft/util/collection/WeightedList.mapping:48]: These methods are automapped

File: mappings/net/minecraft/world/biome/source/CheckerboardBiomeSource.mapping [hunk 4]
```diff
@@ -1,6 +1,8 @@
 CLASS net/minecraft/class_1973 net/minecraft/world/biome/source/CheckerboardBiomeSource
 	FIELD field_24715 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_24716 scale I
 	FIELD field_9480 gridSize I
 	FIELD field_9481 biomeArray Ljava/util/List;
 	METHOD <init> (Ljava/util/List;I)V
+		ARG 1 biomes
 		ARG 2 size
```
Comment (YanisBft) [inline mappings/net/minecraft/world/biome/source/CheckerboardBiomeSource.mapping]: Should match the fields

File: mappings/net/minecraft/world/gen/carver/Carver.mapping [hunk 5]
```diff
@@ -47,6 +48,7 @@ CLASS net/minecraft/class_2939 net/minecraft/world/gen/carver/Carver
 		ARG 1 state
 	METHOD method_12710 getBranchFactor ()I
 	METHOD method_12711 isRegionUncarvable (Lnet/minecraft/class_2791;IIIIIIII)Z
+		ARG 1 chunkX
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/carver/Carver.mapping]:
```suggestion
		ARG 1 chunk
```

File: mappings/net/minecraft/world/gen/decorator/Decoratable.mapping [hunk 6]
```diff
@@ -24,3 +24,5 @@ CLASS net/minecraft/class_5432 net/minecraft/world/gen/decorator/Decoratable
 		COMMENT the input positions by a random number between 0 and {@code maxCount}.
 		ARG 1 maxCount
 			COMMENT the maximum repetition count
+	METHOD method_30377 rangeOf (I)Ljava/lang/Object;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/decorator/Decoratable.mapping:27]: Mind add javadocs like the other methods of the class?

File: mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping [hunk 7]
```diff
@@ -17,21 +17,23 @@ CLASS net/minecraft/class_4647 net/minecraft/world/gen/foliage/FoliagePlacer
 		ARG 8 leaves
 		ARG 9 offset
 		ARG 10 box
-	METHOD method_23449 generate (Lnet/minecraft/class_3747;Ljava/util/Random;Lnet/minecraft/class_4643;Lnet/minecraft/class_2338;ILjava/util/Set;IZLnet/minecraft/class_3341;)V
+	METHOD method_23449 generateSquare (Lnet/minecraft/class_3747;Ljava/util/Random;Lnet/minecraft/class_4643;Lnet/minecraft/class_2338;ILjava/util/Set;IZLnet/minecraft/class_3341;)V
+		COMMENT Generates a square of leaves with the given radius. Sub-classes can use the method isInvalidForLeaves to exclude certain positions, such as corners.
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping]:
```suggestion
		COMMENT Generates a square of leaves with the given radius. Sub-classes can use the method {@code isInvalidForLeaves} to exclude certain positions, such as corners.
```

File: mappings/net/minecraft/world/gen/surfacebuilder/AbstractNetherSurfaceBuilder.mapping [hunk 8]
```diff
@@ -1 +1,16 @@
 CLASS net/minecraft/class_5164 net/minecraft/world/gen/surfacebuilder/AbstractNetherSurfaceBuilder
+	FIELD field_23920 seed J
+	FIELD field_23921 surfaceNoises Lcom/google/common/collect/ImmutableMap;
+	FIELD field_23922 underLavaNoises Lcom/google/common/collect/ImmutableMap;
+	FIELD field_23923 shoreNoise Lnet/minecraft/class_3537;
+	METHOD method_27129 getSurfaceStates ()Lcom/google/common/collect/ImmutableList;
+	METHOD method_27130 (IIILjava/util/Map$Entry;)Ljava/lang/Double;
+		ARG 3 entry
+	METHOD method_27131 createNoisesForStates (Lcom/google/common/collect/ImmutableList;J)Lcom/google/common/collect/ImmutableMap;
+		ARG 0 list
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/surfacebuilder/AbstractNetherSurfaceBuilder.mapping]:
```suggestion
		ARG 0 states
```

File: mappings/net/minecraft/world/gen/tree/TreeDecorator.mapping [hunk 9]
```diff
@@ -2,19 +2,23 @@ CLASS net/minecraft/class_4662 net/minecraft/world/gen/tree/TreeDecorator
 	COMMENT Tree decorators can add additional blocks to trees, such as vines or beehives.
 	FIELD field_24962 TYPE_CODEC Lcom/mojang/serialization/Codec;
 	METHOD method_23469 generate (Lnet/minecraft/class_5281;Ljava/util/Random;Ljava/util/List;Ljava/util/List;Ljava/util/Set;Lnet/minecraft/class_3341;)V
+		COMMENT
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/tree/TreeDecorator.mapping]: Empty comment

File: mappings/net/minecraft/world/gen/trunk/LargeOakTrunkPlacer.mapping [hunk 10]
```diff
@@ -5,21 +5,30 @@ CLASS net/minecraft/class_5212 net/minecraft/world/gen/trunk/LargeOakTrunkPlacer
...
 		ARG 8 config
 	METHOD method_27394 getLongestSide (Lnet/minecraft/class_2338;)I
 		ARG 1 offset
 	METHOD method_27395 getLogAxis (Lnet/minecraft/class_2338;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2350$class_2351;
 		ARG 1 branchStart
 		ARG 2 branchEnd
+	METHOD method_27396 shouldGenerateBranch (II)F
+		COMMENT If the returned value is greater than or equal to 0, a branch will be generated.
+		ARG 1 trunkHeight
+		ARG 2 yy
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/trunk/LargeOakTrunkPlacer.mapping]:
```suggestion
		ARG 2 y
```

File: mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping [hunk 11]
```diff
@@ -13,6 +13,36 @@ CLASS net/minecraft/class_5141 net/minecraft/world/gen/trunk/TrunkPlacer
...
+		ARG 7 config
 	METHOD method_26993 getHeight (Ljava/util/Random;)I
 		ARG 1 random
+	METHOD method_27400 setToDirt (Lnet/minecraft/class_3747;Lnet/minecraft/class_2338;)V
+		ARG 0 world
+		ARG 1 pos
+	METHOD method_27401 trySetState (Lnet/minecraft/class_3747;Ljava/util/Random;Lnet/minecraft/class_2338$class_2339;Ljava/util/Set;Lnet/minecraft/class_3341;Lnet/minecraft/class_4643;)V
+		ARG 0 world
+		ARG 1 random
+		ARG 2 mutable
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping]:
```suggestion
		ARG 2 pos
```

File: mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping [hunk 12]
```diff
@@ -13,6 +13,36 @@ CLASS net/minecraft/class_5141 net/minecraft/world/gen/trunk/TrunkPlacer
...
+		ARG 3 placedStates
+		ARG 4 pos
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping]:
```suggestion
		ARG 4 box
```

File: mappings/net/minecraft/util/collection/WeightedList.mapping (hunk 3 above)
Comment (jaskarth) [reply mappings/net/minecraft/util/collection/WeightedList.mapping:48]: I wanted to map the lambda parameters :P It makes looking at the code much better imo

**Comment by YanisBft** [comment]
@SuperCoder7979 mind resolve the 3 remaining comments please?

**Review by YanisBft — APPROVED** [review]

**Review by Gegy — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**